### DISCUSSION TITLE
What should we do when the chiseled image does not have the necessary permissions in an external volume?

### DISCUSSION THREAD

**Discussion by 64J0** [dotnet/dotnet-docker#4775] [Q&A]
I'm currently facing a problem where the application container must have access to an external volume. It needs to have both read and write permissions for the application to work properly.

When this application tries to check the volume, it throws an error related to the lack of permissions.

How should we handle this, considering the .NET chiseled best practices?

I was considering to change the permissions of the volume folder, but I'm unsure if this is the right path to follow. My main concern is related to the trust in the non-root user of the chiseled image. I was trying to get its information by using the `id` command, but this tool is not available in the chiseled image.

If you guys can share how can I consult the UID and GID of the non-root user of a chiseled image, that would be cool as well.

**Comment by 64J0** [comment]
There's a way to get the user by inspecting the container. One can use this command:

```
# for example this tag jammy-chiseled:
docker pull mcr.microsoft.com/dotnet/nightly/aspnet:6.0-jammy-chiseled
docker image inspect mcr.microsoft.com/dotnet/nightly/aspnet:6.0-jammy-chiseled
```

There we can check the user (ID):

![image](https://github.com/dotnet/dotnet-docker/assets/50725287/66406036-d443-4951-8fce-c24778067400)

Can we trust that this UID is going to be used in next versions?

**Reply by richlander** [reply]
We plan to never change this UID (or name).

**Reply by richlander** [reply]
Typically, the mount needs to give permissions to this specific user.

**Reply by richlander** [reply]
BTW: We should write guidance on this. Feel free to file an issue requesting that.

**Reply by 64J0** [reply]
Cool, created it here: https://github.com/dotnet/dotnet-docker/issues/4778

**Reply by 64J0** [reply]
I'm considering this hacky workaround: https://pratikpc.medium.com/use-docker-compose-named-volumes-as-non-root-within-your-containers-1911eb30f731

**Comment by jander-msft** [comment]
A possible option, if you are using Kubernetes, is to specify a security context with the fsGroup property: https://kubernetes.io/docs/tasks/configure-pod-container/security-context/#set-the-security-context-for-a-pod

This will allow the container processes within the pod to have a supplemental group identifier (whichever value you specified) and the group owner of the volume will be changed to that same identifier. This would possibly allow you to not have to interrogate/change the uid of any container processes or files on the volume.

**Reply by 64J0** [reply]
I'll keep this in mind, thanks for the reply @jander-msft. I'm currently trying to solve this problem for local `docker-compose.yml` only, but the next step is going to be using a Kubernetes local emulator (`minikube` probably).
